### PR TITLE
23 fake data

### PR DESCRIPTION
**Pull request by DeenaAlghazali**
You can use `npm run db:fake` to create fake data

### CODE REVIEW

File: server/database/fakeData/categoryData.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+const allCategories = [
+  {
+    name: 'Food',
+  },
+  {
+    name: 'Money',
+  },
+  {
+    name: 'Clothes',
+  },
```
Comment (Raghad-Mezied) [inline server/database/fakeData/categoryData.ts:12]: these are not a category, change them so they don't lead to a misunderstanding. and you can just name it `categories` without all.

File: server/database/fakeData/campaignsData.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import { Campaign } from '../models/index';
+
+const allCampaign = [
+  {
+    title: 'Helping poor families',
+    description: 'This campaign helps save an amount of money that guarantees 50 families for two months',
+    target: 50000,
+    image_link: 'https://media.voltron.alhurra.com/Drupal/01live-116/styles/sourced/s3/2019-12/AFC8DF4B-8C6D-4968-87B2-CEAFD63DED97.jpg?itok=Y3YypJNm',
+    is_available: true,
```
Comment (Raghad-Mezied) [inline server/database/fakeData/campaignsData.ts:9]: What about the category?

File: server/database/fakeData/categoryData.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { Category } from '../models/index';
+
+const allCategories = [
+  {
+    name: 'Food',
+  },
```
Comment (Raghad-Mezied) [inline server/database/fakeData/categoryData.ts:6]: I think there is an icon too.

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

File: server/database/fakeData/index.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+import donationData from './donationData';
+import caponData from './caponData';
+
+campaignsData();
+categoryData();
+familyData();
+contactData();
+donorData();
+donationData();
+caponData();
```
Comment (Raghad-Mezied) [inline server/database/fakeData/index.ts:15]: Why not create just one function to add all your fake data.

File: server/database/fakeData/contactData.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import { Contact } from '../models/index';
+
+const allMessages = [
+  {
+    name: 'Ahmed',
+    email: 'Ahmed@gmail.com',
+    message: 'Suggest an idea to donate to unemployed people',
+  },
```
Comment (Raghad-Mezied) [inline server/database/fakeData/contactData.ts:8]: I prefer to move all fake data to a separated `json` file.

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

**Comment by Raghad-Mezied** [comment]
Good job Deena, great commits and description.

File: server/database/fakeData/data.json [hunk 6]
```diff
@@ -0,0 +1,201 @@
+{
+  "campaigns": [
+    {
```
Comment (Raghad-Mezied) [inline server/database/fakeData/data.json:3]: This is better than before, but I need you to create a folder for your fake data and add `json` file to each table separately.

File: package.json [hunk 7]
```diff
@@ -11,7 +11,8 @@
     "lint": "eslint --ext .ts server/",
     "lint-fix": "eslint --ext .ts server/ --fix && cd client && npm run lint-fix",
     "lint:client": "cd client && npm run lint",
-    "build:db":"NODE_ENV=dev ts-node server/database/config/build.ts"
+    "build:db": "NODE_ENV=dev ts-node server/database/config/build.ts",
+    "db:fake": "NODE_ENV=dev ts-node server/database/fakeData/index.ts --resolveJsonModule"
```
Comment (Raghad-Mezied) [inline package.json:15]: You can add a short explanation in the description of this script and for `--resolveJsonModule`.

File: server/database/fakeData/index.ts [hunk 8]
```diff
@@ -0,0 +1,48 @@
+import sequelize from '../config/connection';
+// eslint-disable-next-line import/extensions
```
Comment (Raghad-Mezied) [inline server/database/fakeData/index.ts:2]: Do you still need this after editing the eslint rules ?

File: server/database/fakeData/index.ts [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+  Family,
+  Donation,
+  Contact,
+  Category,
+  Capon,
+} from '../models';
+
+const {
+  campaigns, capons, categories, messages, donations, donors, families,
+} = data;
```
Comment (Raghad-Mezied) [inline server/database/fakeData/index.ts:17]: If you are declaring several variables then don't put them in one long line like this, rewrite them like this
```js
const {
campaigns, 
capons, 
categories, 
messages, 
donations, 
donors, 
families,
} = data;
```

File: server/database/fakeData/index.ts [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+
+const buildData = async () => {
+  await sequelize.sync({ force: true });
+
+  if (process.env.NODE_ENV === 'dev') {
```
Comment (Raghad-Mezied) [inline server/database/fakeData/index.ts:22]: What if the env is not a development? You still need to build your fake data. The difference is when you are in a development you run your script to build the data but the script just reads this file which we define a function but we don't use it so we need to invoke it, whereas in the test env we invoke this function in our test files so we don't need to invoke it here again, that why you need to move this condition to the line where you invoke the function.

File: server/database/fakeData/index.ts [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+  Contact,
+  Category,
+  Capon,
+} from '../models';
+
+const {
+  campaigns, capons, categories, messages, donations, donors, families,
+} = data;
+
+const buildData = async () => {
```
Comment (Raghad-Mezied) [inline server/database/fakeData/index.ts:19]: I think it's better to rename the file name to something that shows that we are building the fake data here.

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

**Review by Raghad-Mezied — APPROVED** [review]

**Review by Mohammed-Hamada — CHANGES_REQUESTED** [review]

**Review by Mohammed-Hamada — APPROVED** [review]
Great work deena, you made big effort, Thanks a lot..

**Review by farah2003 — APPROVED** [review]

**Review by HaniOlwan — APPROVED** [review]

**Pull request merged**